src/main/java/org/cactoos/collection/Immutable.java

```diff
@@ -70,7 +71,7 @@ public boolean isEmpty() {
 
     @Override
     public Iterator<X> iterator() {
-        return this.col.iterator();
+        return new IterableOf<X>(this.col.iterator()).iterator();
```

Comment:
@rocket-3 please use `iterator.Immutable` here since we must return an immutable iterator.